Formula/n/neovim.rb

```diff
@@ -148,6 +148,7 @@ def install
     system "cmake", "-S", ".", "-B", "build",
                     "-DLUV_LIBRARY=#{Formula["luv"].opt_lib/shared_library("libluv")}",
                     "-DLIBUV_LIBRARY=#{Formula["libuv"].opt_lib/shared_library("libuv")}",
+                    "-DLPEG_LIBRARY=#{Formula["lpeg"].opt_lib}/lua/5.1/lpeg.so",
```

Comment:
This should use the same `shared_library` function

Reply:
AFAIK, the `shared_library` function [only provides paths to `dylib`](https://github.com/Homebrew/brew/blob/12c8778af976269344f3c0b914243e178fc4ffbd/Library/Homebrew/formula.rb#L1730) libraries and this is an `.so` file. I searched through the rest of the repo and I couldn't find an example for how to use `shared_library` for a `.so` file.

For reference, I ran the following through IRB:
```ruby
>> Formula["lpeg"].opt_lib/shared_library("lpeg")
=> #<Pathname:/usr/local/opt/lpeg/lib/lpeg.dylib>

>> Formula["lpeg"].opt_lib/shared_library("lua/5.1/lpeg")
=> #<Pathname:/usr/local/opt/lpeg/lib/lua/5.1/lpeg.dylib>

>> Formula["lpeg"].opt_lib/shared_library("lua/5.1/lpeg.so")
=> #<Pathname:/usr/local/opt/lpeg/lib/lua/5.1/lpeg.so.dylib>
```

In all of these cases the `shared_library` function is a non-existent library / the wrong file. So I'm wondering what the recommendation here should be.

Additional reference, here's the layout of the `lpeg` library:
```
/usr/local/opt/lpeg/
├── HISTORY
├── INSTALL_RECEIPT.json
├── README.md
├── lib
│   └── lua
│       ├── 5.1
│       │   └── lpeg.so
│       └── 5.4
│           └── lpeg.so
└── share
    ├── doc
    │   └── lpeg
    │       ├── lpeg.html
    │       └── re.html
    ├── lpeg
    │   ├── re.lua
    │   └── test.lua
    └── lua
        ├── 5.1
        │   └── re.lua -> ../../lpeg/re.lua
        └── 5.4
            └── re.lua -> ../../lpeg/re.lua
```

Reply:
on macOS it's the `dylib`, on linux it is a `so` file. If lpeg is building a `.so` on macOS that should be fixed instead.

Reply:
I can fix/update `lpeg` as well if that's the case. But that leaves me with 2 questions:
1. Should I fix that library as part of this PR, or should I file a separate one and rebase this PR after that one is merged.
2. Considering the `lpeg` update is a breaking change to existing dependent users, can you point to an example of a similar change in the past that I can mirror in the `lpeg` update?

Reply:
Should probably be a separate PR and any dependents in Homebrew will show up immediately as a failing linkage check. Anything outside of Homebrew we don't have a mechanism for.

Reply:
`lpeg` looks already fixed via 1a78e1d76a8bcbcd9ce8b62dc47ffa8143c97728

In my the layout of the `lpeg` library is the following:

```
/opt/homebrew/opt/lpeg/
├── HISTORY
├── INSTALL_RECEIPT.json
├── README.md
├── lib
│   ├── liblpeg.dylib -> lua/5.1/lpeg.so
│   └── lua
│       ├── 5.1
│       │   └── lpeg.so
│       └── 5.4
│           └── lpeg.so
└── share
    ├── doc
    │   └── lpeg
    │       ├── lpeg.html
    │       └── re.html
    ├── lpeg
    │   ├── re.lua
    │   └── test.lua
    └── lua
        ├── 5.1
        │   └── re.lua -> ../../lpeg/re.lua
        └── 5.4
            └── re.lua -> ../../lpeg/re.lua
```

Reply:
I added `-DLPEG_LIBRARY=#{Formula["lpeg"].opt_lib/shared_library("liblpeg")}` and successfully installed the neovim HEAD.
And also, I confirmed that `vim.lpeg` is available.